pkg/registry/clusterrolebinding/policybased/storage.go

```diff
@@ -0,0 +1,59 @@
+/*
```

Comment:
We didn't think that was necessary, we were really only interested in preventing escalation. "ReadOnly+Role management" is a weird mix of permissions. 

Separating escalation checks into admission allows them to be accidentally bypassed by anything working directly against the storage once inside the admission boundary. That was scary enough to make us bake it into storage.